api/agent/runner_client.go

```diff
@@ -346,12 +346,24 @@ func recordFinishStats(ctx context.Context, msg *pb.CallFinished, c pool.RunnerC
 	}
 }
 
+func cloneHeaders(src http.Header) http.Header {
+	dst := make(http.Header, len(src))
+	for k, vs := range src {
+		for _, v := range vs {
+			dst.Add(k, v)
+		}
+	}
+	return dst
+}
+
 func receiveFromRunner(ctx context.Context, protocolClient pb.RunnerProtocol_EngageClient, runnerAddress string, c pool.RunnerCall, done chan error) {
 	w := c.ResponseWriter()
 	defer close(done)
 
 	log := common.Logger(ctx).WithField("runner_addr", runnerAddress)
 	statusCode := int32(0)
+	// Make a copy of header to avoid concurrent read/write error when logCallFinish runs.
+	header := cloneHeaders(w.Header())
```

Comment:
can you please rename this "heade" in "clonedHeader" or something similar? That clashes with the name defined in the loop at 386, it doesn't create any issue because the header at line 386 is a different type but It creates some confusion for the reader

Reply:
done